main.rb

```diff
@@ -0,0 +1,93 @@
+
+module Enumerable
+    def my_each
+        for item in self
+            yield(item)
+        end
+    end
+
+  def my_each_with_index
+      for item_index in (0...self.size)
+            yield(self[item_index], item_index)
+      end
+    end
+
+  def my_select
+        new_items = []
+        my_each do |item|
+          new_items.push(yield(item))
+        end
+        new_items
+  end
+
+  def my_all?
+    items_exist = true
+      my_each do |item|
+            friends_exist = false if !yield(item)
+      end
+  items_exist
+
+end
+
+    def my_any?
+        any_item = false
+        my_each do |item|
+            any_item = true if yield(item)
+        end
+        any_item
+    end
+
+    def my_none?
+        item_none = false
+        my_each do |item|
+            item_none = true if !yield(item)
+        end
+        item_none
+    end
+
+    def my_count
+        if block_given?
+        count = 0
+        my_each do |item|
+            count+=1 if yield(item)
+        end
+        count
+    end
+
+
+    def my_map(&proc)
+        new_array = []
+              
+        for item_index in (0...self.size)
+            if block_given?
+                new_array.push(yield(self[item_index]))
+            else
+                new_array.push(proc.call(self[item_index]))
+            end
+        end
+        new_array
+    end
+
+    def my_inject(start = 0)
+       sum = start
+       for item_index in (0...self.size)
+            sum = yield(accumulator, self[item_index])
+       end
+       sum
+    end
+
+    def multiply_els
+    array.my_inject(1) { |x, y| x * y }
+    end
+
+end
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- You have a variable called `accumulator` that is not defined.